### DISCUSSION TITLE
Is it safe to compile parsers with optimizations?

### DISCUSSION THREAD

**Discussion by Colonial-Dev** [tree-sitter/tree-sitter#2665] [Q&A] [answered]
I maintain a library based on `tree-sitter` ([Inkjet](https://github.com/Colonial-Dev/inkjet/)), which bundles the parser sources and compiles/links them at build time.

However, it turns out that I set the `-O2` flag without really thinking in the build script, which the "blessed" Rust bindings do *not* do. On paper, this is fine, and I haven't run into any issues - but better safe than sorry, right?

**Comment by amaanq** [comment, marked as answer]
It's definitely fine, but compiling large lexer functions *can* be slow, and there isn't *much* to optimize anyways
